overlord/overlord.go

```diff
@@ -284,9 +299,18 @@ func (o *Overlord) Loop() {
 // Stop stops the ensure loop and the managers under the StateEngine.
 func (o *Overlord) Stop() error {
 	o.loopTomb.Kill(nil)
-	err1 := o.loopTomb.Wait()
+	finalErr := o.loopTomb.Wait()
 	o.stateEng.Stop()
-	return err1
+	if o.udevMon != nil {
+		if err := o.udevMon.Stop(); err != nil {
+			logger.Noticef("Failed to stop udev monitor: %s", err)
+			if finalErr == nil {
```

Comment:
Yes, that's a valid point. The reason it's done like that at the moment is the fact that we don't have a way to "start" and "stop" the managers - in this case we need to start the udev monitor subroutine when everything else is ready, and stop it on shutdown. 
Do we want `Start` and `Stop` methods on the managers to cater for this use case to make the concept abstract. or - if all this is moved under ifacemgr, just special case for interface manager?

Reply:
We always had and still support a (optional now) Stop method on managers, so it would be a matter of adding Start

Reply:
Fwiw, I played with making udevmonitor a manager in https://github.com/mvo5/snappy/commit/bf4182147717a5fa08359d69b8bb03a0851a3eca - this implements an optional "Start" for managers (if its needed it could be taken from there).

Reply:
Ok, never mind, completely missed the optional `StateWaiterStopper` defined in stateengine, that managers can implement (that is, Stop and Wait method). Together with Ensure it makes sense now, thank you!

Reply:
@mvo5 Thank you for experimenting with it! At the moment I've implement hotplug start inside Ensure as it needs to be retried (I retry every few minutes - TBD) on failure, per the discussion from our standup., but Start certainly might have its uses, so perhaps worth proposing anyway as a separate PR?

Reply:
I suggest not implementing a Start, at least not yet. The reason is exactly the one you described: Ensure encourages a pattern of retrying failures, and reevaluating state correctness.